pubtools/pulplib/_impl/client/client.py

```diff
@@ -203,6 +203,33 @@ def search_repository(self, criteria=None):
             Repository, "repositories", criteria=criteria, search_options=search_options
         )
 
+    def search_units_by_type(self, content_type, criteria=None):
```

Comment:
This API doesn't seem to make sense when taken together with the existing Repository.search_content.

- Why is Repository.search_content named search_content but this is named search_units* ?
- Why does this API require the caller to pass content_type as a separate argument while Repository.search_content instead allows putting content_type_id into Criteria?  

To elaborate on the second point:

```
# If I have a repo, I can find all units of type "foo" matching certain criteria like this:
repo.search_content(Criteria.and_(Criteria.with_field("content_type_id", "foo"), <other_criteria>))

# But if I want to search across *all* repos I instead need to do:
client.search_units_by_type("foo", <other_criteria>)
```

Why the inconsistency?  There doesn't seem to be any reason why per-repo search and all-repo search methods need to have different signatures.

Reply:
What if you pass there criteria like:
Criteria.with_field("content_type_id", Matcher._in(["rpm","srpm"]))
Which content_type_id should be then used for the search?
That's why I rather made it inconsistent.

Reply:
It should then do more than one request for the search.  It's not much different from pagination - it's already undefined how many requests are done to the server for any particular search.

Reply:
ok, but what if user specify criteria like 
`
Criteria.with_field("content_type_id", Matcher._regex("?rpm"))
`
It's most probably not supported right now, but it could be in the future if TypeIdsAccumulator.accumulate_from_match changes.
That would basically mean, I need to gather available content types from server and do the matching locally. And that I wanted to avoid.
Reasonable could be subsclassing TypeIdsAccumulator with restriction only for matchers like $eq and $in.
I would like to hear your preferred solution for this before I got o start changing anything.

Reply:
That specific example should give them a fatal error as not supported (before the request is sent to the server).  Support can be added later if there is a use-case for it.

My preferred solution is to discourage searching on content_type_id at all and let the caller instead pass the model class they're interested in, since the class already knows its associated Pulp unit types and this info ideally should not have to be duplicated by the caller.  As written in https://github.com/release-engineering/pubtools-pulplib/issues/64 :

```
unit = RpmUnit(name='bash')
crit = Criteria.for_units([unit])
# Search all bash RPMs.
# Under the hood, the library knows that RpmUnit can belong to 'rpm' or 'srpm' content types
# and so it issues search requests for both.
client.search_content(criteria=crit)
```

However I don't think this PR should be forced to implement that, a "raw" search on content_type_id is OK as long as it's consistent with repo.search_content.